pkg/github/server.go

```diff
@@ -29,6 +30,27 @@ func NewServer(version string, opts ...server.ServerOption) *server.MCPServer {
 	return s
 }
 
+type constrainableValue interface {
+	Constrain(any) error
+}
+
+type constrainableInt32 int32
+
+func (ci *constrainableInt32) Constrain(param any) error {
```

Comment:
Ok, I'm not sure if this is way overengineered or not, but it facilitates constraining param values: https://github.com/github/github-mcp-server/pull/381/files#diff-c2e2a7250ef597d08ee5d429159bc89c4b862a7ed6cbcf30b8231a3ec599d190R1081

Reply:
I will get an error calling this as a lib right? Because the type and interface are not exported so I cannot provide them.

I don't need to call it I suppose, just a thought that I would need to do that if I need to add similar functionality.

To your question, I think it makes sense - we want to avoid integer wrapping, and sanitising user input is generally a good thing. It is a little heavy handed perhaps, but I expect we will not touch it often. I've not had to touch any of these functions much, they largely just do what you'd expect, so I don't mind it - it's not a huge footprint.

Reply:
I suppose one comment might be, why not allow the caller to pass the min and max, in case they wish to constrain a value futher? Then indeed you could lose the interface, and have a generic function with numeric argument with a min and max check based on user provided value?

🤔 hard to decide. I do get the desire for extending the type directly.

Reply:
I'm just going to remove it in favour of `ParseInt32`. It's simpler, and if we later discover we want this behaviour, we have evidence for it.